### PR TITLE
[Merged by Bors] - fix: Enabled `--file` and `--key-separator` to be used together, fix `--key` handling when producing lines

### PR DESCRIPTION
**Pull request by ozgrakkurt**
closes https://github.com/infinyon/fluvio/issues/3076

The issue also mentions it should be added to the CI. I'm looking into doing that.

DISCLAIMER: I didn't test this yet

### CODE REVIEW

**Comment by ozgrakkurt** [comment]
Added test

**Comment by ozgrakkurt** [comment]
It seems like this is failing because of something with CI, is it not rebuilding the cli?

**Comment by EstebanBorai** [comment]
> ~It seems like this is failing because of something with CI, is it not rebuilding the cli?~
> 
> nevermind I see the point

Hi @ozgrakkurt! Good work so far!

Looks like the issue is here: https://github.com/infinyon/fluvio/actions/runs/4504419808/jobs/7929176514?pr=3092#step:16:72

Due to conflicts with arguments passed on test.

**Comment by ozgrakkurt** [comment]
Hey @EstebanBorai,

Old version fails with those options and part of the PR was making it so that it accepts those options.

So the test I added fails with old version but passes with new version.

Not sure how to integrate this with the test.

**Comment by sehz** [comment]
In that case, run that particular test on new version

**Comment by ozgrakkurt** [comment]
@sehz  I added some hack to do this in the last commit. What would be a proper way of doing this?

**Comment by sehz** [comment]
That's reasonable fix!

**Review by EstebanBorai — APPROVED** [review]
LGTM! Nice work @ozgrakkurt!

**Comment by ozgrakkurt** [comment]
@EstebanBorai can you re-check when you have time?

**Comment by EstebanBorai** [comment]
@ozgrakkurt smart way to tune tests for this use case!

**Comment by ozgrakkurt** [comment]
@EstebanBorai thanks!

File: tests/cli/fluvio_smoke_tests/e2e-file-with-separator.bats [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+teardown_file() {
+    run timeout 15s "$FLUVIO_BIN" topic delete "$TOPIC_NAME"
+    run rm $MULTI_LINE_FILE_NAME
+}
+
+# Create topic
+@test "Create a topic for file message with separator" {
+    if [ "$FLUVIO_CLI_RELEASE_CHANNEL" == "stable" ]; then
+        skip "don't run on stable version"
+    fi
```
Comment (morenol) [inline tests/cli/fluvio_smoke_tests/e2e-file-with-separator.bats:50]: nice trick!

We should create an issue to update this once we release a new stable version

Comment (ozgrakkurt) [reply tests/cli/fluvio_smoke_tests/e2e-file-with-separator.bats:50]: thanks!

**Review by morenol — APPROVED** [review]

**Comment by ozgrakkurt** [comment]
Hey @sehz, can you check when you have time?

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**